### PR TITLE
Bump dependencies

### PR DESCRIPTION
**Pull request by gsanchezgavier**
### Changed
- Update Go version to 1.18
- Bump dependencies
### Added
Added support for more distributions:
- RHEL(EL) 9
- Ubuntu 22.04
- Amazon Linux 2022

### CODE REVIEW

File: src/client/client.go [hunk 1]
```diff
@@ -35,7 +35,18 @@ func CreateClient(args *arguments.ArgumentList, hostnameOverride string) (*HTTPC
 		return args.Hostname
 	}()
 
-	httpClient, err := nrHttp.New(args.CABundleFile, args.CABundleDir, time.Duration(args.Timeout)*time.Second)
+	options := []nrHttp.ClientOption{
```
Comment (paologallinaharbur) [inline src/client/client.go:38]: is this due to the sdk change?

File: tools/go.mod [hunk 2]
```diff
@@ -0,0 +1,165 @@
+module github.com/newrelic/nri-couchbase/tools
+
+go 1.18
+
+require github.com/golangci/golangci-lint v1.46.2
```
Comment (paologallinaharbur) [inline tools/go.mod:5]: I have a doubt.. who is using this? Arent we leveraging the action https://github.com/newrelic/nri-couchbase/blob/032dd8c92c0007eab3001cff3207c6869c650222/.github/workflows/push_pr.yml#L28?

I noticed it in different repos

File: src/client/client.go (hunk 1 above)
Comment (gsanchezgavier) [reply src/client/client.go:38]: yes, the option pattern was introduced to it, and the With.. methods fails if the CA fields are empty, that is why the conditional instantiation

File: tools/go.mod (hunk 2 above)
Comment (gsanchezgavier) [reply tools/go.mod]: was discussed I have removed all local validate related stuff

**Review by paologallinaharbur — APPROVED** [review]
🔥

**Pull request merged**